tests/examples/test_dataset_loader.py

```diff
@@ -69,10 +69,8 @@ def add_to_dict(func: str, dataset_function: Callable[[], Any]):
 
     # Collect all `download_<name> or `load_<name>` functions
     def _is_dataset_function(name, item):
-        return (
-            isinstance(item, FunctionType)
-            and name.startswith('download_')
-            or name.startswith('load_')
+        return (isinstance(item, FunctionType) and name.startswith('download_')) or name.startswith(
+            'load_'
         )
```

Comment:
This one looks to be blatantly incorrect. The check is first evaluating `isinstance(item, FunctionType)` and if so see if the name starts with download or load.

This makes me wearing of all changes to conditional checks in this PR

Reply:
In my reading of https://docs.python.org/3/reference/expressions.html#operator-precedence ruff is making the order of python precedence explicit.  That is, it evaluates the `and` first then the `or` second.  So the ruff version is giving us what is currently being done, but explicitly showing it.  Which means that the code was incorrect and ruff found a bug.

Reply:
This one is a bug.